workbench-core/datasets/src/wbcDataSetsAuthorizationPlugin.ts

```diff
@@ -31,30 +31,41 @@ export class WbcDataSetsAuthorizationPlugin implements DataSetsAuthorizationPlug
     }
 
     if (params.permission.identityType !== 'GROUP' && params.permission.identityType !== 'USER') {
-      throw new InvalidPermissionError('IdentityType just be "GROUP" or "USER".');
+      throw new InvalidPermissionError('IdentityType must be "GROUP" or "USER".');
     }
 
     const permissionsEffect: Effect = 'ALLOW';
-    const permissionsAction: Action = params.permission.accessLevel === 'read-only' ? 'READ' : 'UPDATE';
     const subjectType: IdentityType = params.permission.identityType === 'GROUP' ? 'GROUP' : 'USER';
 
+    const identityPermissions: IdentityPermission[] = [
+      {
+        identityType: subjectType,
+        identityId: params.permission.identity,
+        action: 'READ',
+        effect: permissionsEffect,
+        subjectType: 'DataSet',
+        subjectId: params.dataSetId,
+        description: `${params.permission.accessLevel} access on DataSet ${params.dataSetId}`
+      }
+    ];
+    if (params.permission.accessLevel === 'read-write') {
+      identityPermissions.push({
+        identityType: subjectType,
+        identityId: params.permission.identity,
+        action: 'UPDATE',
+        effect: permissionsEffect,
+        subjectType: 'DataSet',
+        subjectId: params.dataSetId,
+        description: `read-write access on DataSet ${params.dataSetId}`
+      });
+    }
     const createdPermission: CreateIdentityPermissionsResponse =
       await this._authorizer.createIdentityPermissions({
         authenticatedUser: {
           id: params.authenticatedUserId,
           roles: params.roles
         },
-        identityPermissions: [
-          {
-            identityType: subjectType,
-            identityId: params.permission.identity,
-            action: permissionsAction as Action,
-            effect: permissionsEffect as Effect,
-            subjectType: 'DataSet',
-            subjectId: params.dataSetId,
-            description: `${params.permission.accessLevel} on DataSet ${params.dataSetId}`
-          }
-        ]
+        identityPermissions: identityPermissions
```

Comment:
```suggestion
        identityPermissions
```

Reply:
It's hard to see in the code snippet, but the `identityPermissions` you are recommending I remove is not the `IdentityPermissions` type, but an object which contains the parameter values I built in the previous step.

Reply:
yep, when a property and its value share the same name, you can remove the value to make the code cleaner. ie
```
const name = 'John Doe';

interface Person {
  name: string;
  age: number;
}

const someone: Person = {
  name,
  age: 30
}

console.log(someone.name); // prints 'John Doe'
```

Reply:
That's easier to read? Okay. :)